TRT-django/marketplace/views.py

```diff
@@ -465,16 +466,54 @@ def getItemsRelative(request):
         items = items.filter(categories__in=categories)
         items = Item.objects.filter(pk__in=items) # get rid of duplicate rows (can happen because of filtering on m2m categories table)
 
-    # annotate items by search string rank
-    items = items.annotate(rank=SearchRank(SearchVector("name", "description"), SearchQuery(search_string), cover_density=True))
+    # sort items by price or date if requested
+    if "sort_type" in request.GET:
+        sort_type = request.GET["sort_type"]
+    
+    if sort_type == "price_hightolow":
+        items = items.annotate(
+            row=Window(
+                expression=RowNumber(),
+                order_by=[F("price").asc(), F("pk").asc()], # also order by unique pk to make tie-breaks consistent
+            )
+        )
```

Comment:
could you instead use this conditional block only to set the value passed into the `order_by` argument, and then make one call to `items.annotate` at the end?

```python
if sort_type == "price_hightolow":
    order_by = [F("price").asc(), F("pk").asc()]
elif ...:
    order_by = ...
elif ...:
    order_by = ...
elif ...:
    order_by = ...
else: ...
    order_by = ...

    # annotate items by search string rank
    items = items.annotate(rank=SearchRank(SearchVector("name", "description"), SearchQuery(search_string), cover_density=True))

items = items.annotate(
            row=Window(
                expression=RowNumber(),
                order_by=order_by, # also order by unique pk to make tie-breaks consistent
            )
        )
```

Reply:
^ I'd also like to see the duplicate logic for items and item requests combined in some way